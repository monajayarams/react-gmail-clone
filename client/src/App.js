import React, { useState, useEffect } from 'react'
import styled from 'styled-components'
import { HomePage } from './components/HomePage'
import { Login } from './components/Login'

const Conatiner = styled.div`
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  height: 100vh;
  width: 100vw;
  min-height: fit-content;
  min-width: fit-content;
`

const App = () => {
  const [isLoggedIn, setLoggedIn] = useState(false)
  useEffect(() => {
    const login = window.localStorage.getItem('isUserLoggedIn')
    const token = window.localStorage.getItem('bearerToken')
    if(login && token) {
      setLoggedIn(true)
    }
  },[])

  const handleLogout = () => {
    window.localStorage.removeItem('isUserLoggedIn')
    window.localStorage.removeItem('bearerToken')
    setLoggedIn(false)
  }

  const handleLogin = () => {
    setLoggedIn(true)
  }
  return (
    <Conatiner>
      {isLoggedIn && 
        <HomePage handleLogout={handleLogout} /> 
      }
      {!isLoggedIn && 
        <Login handleLoginCallback={handleLogin} />
      }
    </Conatiner>
  )
}

export default App
