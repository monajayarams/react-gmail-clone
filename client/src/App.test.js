import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

describe('verify App Snapshot', () => {
  it('loads app correctly', () => {
    const app = render(<App />)
    expect(app).toMatchSnapshot()
  })
})
