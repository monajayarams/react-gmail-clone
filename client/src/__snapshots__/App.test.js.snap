// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verify App Snapshot loads app correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gSAPjG cHUloM"
      >
        <div
          class="sc-hTtwUo gfIMOy"
        >
          <div
            class="sc-cOFTSb eJPxCc"
          >
            <div
              class="sc-kgUAyh UfxGg"
            >
              <img
                alt="logo"
                src="https://ssl.gstatic.com/ui/v1/icons/mail/rfr/logo_gmail_lockup_dark_1x_r2.png"
              />
              <h3>
                Sign in
              </h3>
              <p>
                to continue to Gmail
              </p>
            </div>
            <form
              class="sc-BeQoi izpmLQ"
            >
              <input
                class="sc-jgbSNz jLSsai"
                placeholder="Email"
                required=""
                type="text"
                value=""
              />
              <input
                class="sc-jgbSNz jLSsai"
                placeholder="Password"
                required=""
                type="password"
                value=""
              />
              <button
                type="submit"
              >
                Login
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gSAPjG cHUloM"
    >
      <div
        class="sc-hTtwUo gfIMOy"
      >
        <div
          class="sc-cOFTSb eJPxCc"
        >
          <div
            class="sc-kgUAyh UfxGg"
          >
            <img
              alt="logo"
              src="https://ssl.gstatic.com/ui/v1/icons/mail/rfr/logo_gmail_lockup_dark_1x_r2.png"
            />
            <h3>
              Sign in
            </h3>
            <p>
              to continue to Gmail
            </p>
          </div>
          <form
            class="sc-BeQoi izpmLQ"
          >
            <input
              class="sc-jgbSNz jLSsai"
              placeholder="Email"
              required=""
              type="text"
              value=""
            />
            <input
              class="sc-jgbSNz jLSsai"
              placeholder="Password"
              required=""
              type="password"
              value=""
            />
            <button
              type="submit"
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
